### PR TITLE
RESTEASY-2203 add example for tracing feature

### PR DESCRIPTION
**Pull request by liweinan**

### CODE REVIEW

**Comment by liweinan** [comment]
@asoldano could you please help to review this? Thanks!

File: pom.xml [hunk 1]
```diff
@@ -13,7 +12,15 @@
...
     <properties>
+        <javax-json.version>1.1</javax-json.version>
+        <jackson-databind.version>2.9.4</jackson-databind.version>
+        <jboss-logging.version>3.3.2.Final</jboss-logging.version>
+        <junit.version>4.12</junit.version>
+        <logmanager.version>2.1.8.Final</logmanager.version>
+        <maven-compiler-plugin.version>3.8.0</maven-compiler-plugin.version>
         <resteasy.version>4.0.0.CR1</resteasy.version>
+        <resteasy-tracing-api.version>1.0.0.Beta4</resteasy-tracing-api.version>
+        <undertow.ver>1.4.18.Final</undertow.ver>
```
Comment (asoldano) [inline pom.xml]: For all the dependencies here, why can't we rely on what RESTEasy would transitively pull? That would reduce the burden of maintaining versions here in future.

File: tracing-example/src/main/java/org/jboss/resteasy/tracing/examples/Demo.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import org.jboss.resteasy.core.ResteasyDeploymentImpl;
+import org.jboss.resteasy.plugins.server.servlet.ResteasyContextParameters;
+import org.jboss.resteasy.plugins.server.undertow.UndertowJaxrsServer;
+import org.jboss.resteasy.spi.ResteasyDeployment;
+
+import java.io.IOException;
+
+public class Demo {
+    public static UndertowJaxrsServer buildServer() {
+        UndertowJaxrsServer Server;
```
Comment (asoldano) [inline tracing-example/src/main/java/org/jboss/resteasy/tracing/examples/Demo.java]: s/Server/server/ please

File: tracing-example/src/test/java/BasicQueryTest.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        WebTarget target = client.target("http://localhost:8081/type");
+        assertEquals(ResteasyContextParameters.RESTEASY_TRACING_TYPE_ALL, target.request().get(String.class));
+
+        target = client.target("http://localhost:8081/level");
+        assertEquals(ResteasyContextParameters.RESTEASY_TRACING_LEVEL_VERBOSE, target.request().get(String.class));
+
+        target = client.target("http://localhost:8081/logger");
+        assertEquals(RESTEasyTracingLogger.class.getName(), target.request().get(String.class));
+
+        if (client != null) {
```
Comment (asoldano) [inline tracing-example/src/test/java/BasicQueryTest.java]: you can remove the check on client being null here, if it was you'd get a NPE few lines above.

File: tracing-example/src/main/java/org/jboss/resteasy/tracing/examples/jackson/Play.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
+package org.jboss.resteasy.tracing.examples.jackson;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.lang.reflect.Method;
+
+public class Play {
+    public static void main(String[] args) throws Exception {
```
Comment (asoldano) [inline tracing-example/src/main/java/org/jboss/resteasy/tracing/examples/jackson/Play.java]: What's this class for? I don't see it mentioned in the doc nor in the code...

File: tracing-example/src/main/java/org/jboss/resteasy/tracing/examples/jsonb/Play2.java [hunk 5]
```diff
@@ -0,0 +1,15 @@
+package org.jboss.resteasy.tracing.examples.jsonb;
+
+import java.lang.reflect.Method;
+
+public class Play2 {
```
Comment (asoldano) [inline tracing-example/src/main/java/org/jboss/resteasy/tracing/examples/jsonb/Play2.java]: What's this class for? I don't see it mentioned in the doc nor in the code...

File: tracing-example/src/main/java/org/jboss/resteasy/tracing/examples/sample.json [hunk 6]
```diff
@@ -0,0 +1,196 @@
+{
```
Comment (asoldano) [inline tracing-example/src/main/java/org/jboss/resteasy/tracing/examples/sample.json]: What's the use of this sample.json file?

File: pom.xml (hunk 1 above)
Comment (liweinan) [reply pom.xml]: yes sir!

File: tracing-example/src/test/java/BasicQueryTest.java (hunk 3 above)
Comment (liweinan) [reply tracing-example/src/test/java/BasicQueryTest.java]: okay!

File: tracing-example/src/main/java/org/jboss/resteasy/tracing/examples/jsonb/Play2.java (hunk 5 above)
Comment (liweinan) [reply tracing-example/src/main/java/org/jboss/resteasy/tracing/examples/jsonb/Play2.java]: removed

File: tracing-example/src/main/java/org/jboss/resteasy/tracing/examples/sample.json (hunk 6 above)
Comment (liweinan) [reply tracing-example/src/main/java/org/jboss/resteasy/tracing/examples/sample.json]: recycled

File: tracing-example/src/main/java/org/jboss/resteasy/tracing/examples/jackson/Play.java (hunk 4 above)
Comment (liweinan) [reply tracing-example/src/main/java/org/jboss/resteasy/tracing/examples/jackson/Play.java]: discarded

File: tracing-example/src/main/java/org/jboss/resteasy/tracing/examples/Demo.java (hunk 2 above)
Comment (liweinan) [reply tracing-example/src/main/java/org/jboss/resteasy/tracing/examples/Demo.java]: okay!

**Comment by liweinan** [comment]
@asoldano all are fixed, could you please help to review it? Thanks!

**Pull request merged**